src/DestroyStrategy.groovy

```diff
@@ -33,6 +33,7 @@ class DestroyStrategy {
             node(jenkinsfile.getNodeName()) {
                 deleteDir()
                 checkout(scm)
+                properties([parameters(params)])
 
```

Comment:
This is making me really uncomfortable from a maintainability perspective.  I don't want to have to update every strategy from now and into the future to make stuff like this work.

Can we approach this differently?  This could be a decoration around ALL, couldn't it?  

^---These statements apply to the DefaultStrategy as well.

Reply:
Hrm, this is the issue with multiple plugins trying to set properties isn't it?

If multiple plugins want to inject properties, they overwrite each other?

Reply:
I'm not sure what you mean by "I don't want to have to update every strategy from now and into the future to make stuff like this work". You wouldn't need to update every strategy going forward.

When you write a new strategy, you would just copy the code from defaultStrategy (which already has the `properties([parameters(params)])` line. 

I had experimented with trying a decoration around ALL. However, I was hitting a race condition where the parameters would not be defined when running the pipeline (since they were appended as parameters after the building of the pipeline)

Reply:
Has this been tested against Declarative pipelines?

Reply:
> Hrm, this is the issue with multiple plugins trying to set properties isn't it?
> 
> If multiple plugins want to inject properties, they overwrite each other?

yes they overwrite. Thats why I wanted to define a list of params which we can append to.

Reply:
> Has this been tested against Declarative pipelines?

No not yet